core/engine/src/context/mod.rs

```diff
@@ -892,8 +898,8 @@ impl Context {
 
     /// Get the Time Zone Provider
     #[cfg(feature = "temporal")]
-    pub(crate) fn tz_provider(&self) -> &impl TimeZoneProvider {
-        &self.tz_provider
+    pub(crate) fn timezone_provider(&self) -> &impl TimeZoneProvider {
```

Comment:
I'd expose this as `&dyn TimeZoneProvider` instead, which avoids the additional generic